keytree/src/lib.rs

```diff
@@ -106,6 +169,42 @@ mod tests {
         assert_eq!(expected_scalar, xprv.scalar);
     }
 
+    #[test]
+    fn serialize_xprv_test() {
+        let seed = [0u8; 32];
+        let mut rng = ChaChaRng::from_seed(seed);
+        let xprv = Xprv::random(&mut rng);
+        let xprv_bytes = xprv.to_bytes();
+
+        // hardcoded, but happens to be expected_scalar concatenated with expected_dk
+        let expected_bytes = [
```

Comment:
there's `hex` crate with `hex::encode` method so we can write a more readable test (and then copy this over to the spec as Test Vectors):

```rust
assert_eq!(hex::encode(&xprv.to_bytes()), "df0da2b88d88de1ad7678d8f012...");
```

Reply:
`hex` is only needed in `dev-dependencies` in Cargo.toml. So you'd include it only under the `mod test { }`, not in the top of lib.rs.

Reply:
I just tried this out, and it's neat, but I'm not sure it actually produces a more readable test, especially since all the other tests in this package (for creating xprvs and xpubs) use non-hex encoded byte arrays. When I switched over to a hex string, I could no longer eyeball that these were the same. 

WDYT?

Reply:
i think all have to be hex, across the board. But if it's a PITA to update, we can do that in a separate PR later.

Reply:
we'd definitely want that to post the test vectors in the spec in hex. But before that we need to implement derivation for both xpub and xprv.

Reply:
I'll do it all in another PR, then. (It's fine to do, I should just change them all together.)